src/abstractdataframe/join.jl

```diff
@@ -280,7 +285,21 @@ join(name, job2, on = :ID => :identifier)
 function Base.join(df1::AbstractDataFrame,
                    df2::AbstractDataFrame;
                    on::Union{<:OnType, AbstractVector{<:OnType}} = Symbol[],
-                   kind::Symbol = :inner, makeunique::Bool=false)
+                   kind::Symbol = :inner, makeunique::Bool=false,
+                   indicator::Union{Nothing, Symbol} = nothing)
+    if indicator !== nothing
+        indicator_cols = ["_left", "_right"]
+        for i in 1:2
+            while (haskey(index(df1), Symbol(indicator_cols[i])) ||
```

Comment:
What was the problem with `c = Symbol(indicator_cols[i])`? It was more readable.

BTW, you can move `df1[Symbol(indicator_cols[1])] = trues(nrow(df1))` and `df2[Symbol(indicator_cols[2])] = trues(nrow(df2))` inside the `for`.

Reply:
The `c = Symbol(indicator_cols[i])` had to be written twice -- once before the loop and once inside the loop again (or the change didn't propagate). DIdn't like have the code duplicated.

Reply:
Putting them inside the `for` requires some tricks to make `df1` and `df2` indexable. I think more readable with just two explicit lines.

Reply:
e.g. had to be:

```
c = Symbol(indicator_cols[i])
while (haskey(index(df1), c) || haskey(index(df2), c) || Symbol(indicator_cols[i]) == c)
    indicator_cols[i] *= 'X'
    c = Symbol(indicator_cols[i])
end
```

or something like that.